.editorconfig

```diff
@@ -0,0 +1,14 @@
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+charset = utf-8
+indent_style = space
+
+[*.rs]
+indent_size = 4
```

Comment:
I like adding `max_line_length = 100` (maybe for all)

I would also like to use the nixfmt-rfc-style nix formatter!

https://github.com/rust-lang/rust/blob/master/.editorconfig

Reply:
Let's do Nix formatting with another PR, but the max line length sounds good, adding it for `[*]` :)